plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -761,6 +762,13 @@ private ObjectMetadata getS3ObjectMetadata(Path path, String bucketName, String
                         }
                         catch (RuntimeException e) {
                             STATS.newGetMetadataError();
+                            // Retry AWS STS Throttling error code specifically since it uses HTTP_BAD_REQUEST status code
+                            // For example: "Caused by: com.amazonaws.services.securitytoken.model.AWSSecurityTokenServiceException: Rate exceeded (Service: AWSSecurityTokenService; Status Code: 400; Error Code: Throttling; ...)
+                            if (e instanceof AWSSecurityTokenServiceException
+                                    && ((AWSSecurityTokenServiceException) e).getStatusCode() == HTTP_BAD_REQUEST
+                                    && ((AWSSecurityTokenServiceException) e).getErrorCode().equals("Throttling")) {
+                                throw e;
+                            }
                             if (e instanceof AmazonServiceException) {
                                 switch (((AmazonServiceException) e).getStatusCode()) {
                                     case HTTP_FORBIDDEN:
```

Comment:
why do you need separate `if` when `AWSSecurityTokenServiceException` is `AmazonServiceException`? `getStatusCode` and `getErrorCode` should be available.

Reply:
I opted for a separate `if` to avoid adding a nested `if` statement (personal preference) in the `switch` statement, and following the example for handling S3 404 errors. e.g. https://github.com/trinodb/trino/blob/e4a1a811353dd3fb223d83db287e6e0612ea8fdc/plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java#L779

Is there some Java syntax I'm missing on handling the throttling error cleanly in that `if` statement?